Core/Src/main.c

```diff
@@ -96,8 +104,33 @@ int main(void)
   while (1)
   {
     /* USER CODE END WHILE */
+	  /* USER CODE BEGIN 3 */
+	  uint8_t transmit_data[]={0x01,0x80,0x00};
+	  uint8_t receive_data[3];
+	  uint16_t adc_value; //received data is 3 bytes, first and second bytes needs to be kept
+	  uint16_t counts;
+
+	  //Bring CS pin to low
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET);
+
+	  //Transmit and receive data via SPI
+	  HAL_SPI_TransmitReceive(&hspi1, transmit_data, receive_data, sizeof(transmit_data), HAL_MAX_DELAY);
+
+	  //Bring CS pin to high
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET);
+
+	  //calculate ADC value
+	  adc_value = (receive_data[1]) << 8;
+	  adc_value = adc_value + (receive_data[2]);
 
-    /* USER CODE BEGIN 3 */
+	  //calculate count
+	  //count=period*cycle + normalized ADC value range
+	  counts = (PERIOD * DUTY_CYCLE_MIN) + ((float)adc_value / ADC_RESOLUTION-1) * PERIOD * (DUTY_CYCLE_MAX - DUTY_CYCLE_MIN);
```

Comment:
Be careful of the precedence of operators. `((float)adc_value / ADC_RESOLUTION-1)` the division happens before minus, and I don't think this is what you want.